package/MDAnalysis/coordinates/chain.py

```diff
@@ -271,6 +271,13 @@ def __init__(self, filenames, skip=1, dt=None, continuous=False,
                         for filename in filenames]
         self.filenames = np.array([fn[0] if isinstance(fn, tuple) else fn
                                                         for fn in filenames])
+        # Set filenames array to array of "None" if chainreader initialized with
+        # numpy arrays
+        if isinstance(filenames[0], np.ndarray):
+            self.filenames = np.array(["None" for _ in range(len(filenames))])
```

Comment:
> If there is a use case for a ChainedReader with both MemoryReader sub-readers and other sub-reader types, this solution won't work because it depends on all the filenames being the same type.

I'm not sure this would be a *common* use-case, but I can definitely concoct a non-implausible scenario where a user wants to pre-pend some frames onto an existing trajectory! Would it be a lot of work to iterate through `filenames` and check if each is a `np.ndarray` or `tuple`, to have it match more closely to the actual underlying readers and be a bit more robust?

Separately, it was a good idea to keep the types the same no matter what, so that `self.filenames` is always an array of strings. However, it might be more natural to use the `NoneType` `None` value here instead, just to be clear that there is no filename vs. potentially having a file named `"None"` 😅 It also helps with simple boolean `if filenames[0]` checks, as `"None"` will evaluate truthily due to be a non-empty string.